tensorflow/java/src/main/native/tensor_jni.cc

```diff
@@ -218,6 +218,45 @@ size_t readNDArray(JNIEnv* env, TF_DataType dtype, const char* src,
 JNIEXPORT jlong JNICALL Java_org_tensorflow_Tensor_allocate(JNIEnv* env,
                                                             jclass clazz,
                                                             jint dtype,
+                                                            jlongArray shape,
+                                                            jlong size) {
+  int num_dims = static_cast<int>(env->GetArrayLength(shape));
```

Comment:
Most of this is duplicated in `allocateNDArray`, perhaps structure this so that the code isn't duplicated (`allocateNDArray` can call `allocate` after determining the byte size). Though, see comment above about `allocate` and `allocateForStrings`.